inspire_json_merger/pre_filters.py

```diff
@@ -113,6 +116,35 @@ def filter_publisher_references(root, head, update):
     return root, head, update
 
 
+def update_authors_with_ordering_info(root, head, update):
+    """Adds ordering information into authors entries.
+
+    This is helpful so we won't loose correct ordering on conflicts.
+    Args:
+        root (pmap): the root record.
+        head (pmap): the head record.
+        update (pmap): the update record.
+
+    Returns: A tuple containing root, head and update with authors data
+        enriched with ordering information
+
+    """
+    if root.get("authors"):
+        root = root.update({"authors": _update_authors_list_with_ordering_data(root["authors"], 0)})
+    if head.get('authors'):
+        head = head.update({"authors": _update_authors_list_with_ordering_data(head["authors"], 1)})
+    if update.get('authors'):
+        update = update.update({"authors": _update_authors_list_with_ordering_data(update["authors"], 2)})
+    return root, head, update
+
+
+def _update_authors_list_with_ordering_data(input_list, second_key):
+    new_list = []
+    for position, item in enumerate(input_list):
+        new_list.append(item.update({ORDER_KEY: (position, second_key)}))
+    return freeze(new_list)
```

Comment:
Better use `transform` here to avoid building a new list and freezing it. Something like
```suggestion
    positions = iter(range(input_list))
    return input_list.transform([pyrsistent.ny], lambda element: element.set(ORDER_KEY, (next(positions), second_key)))
```

Reply:
:+1: